tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -135,13 +135,18 @@ class Radix2EvaluationDomain : public UnivariateEvaluationDomain<F, MaxDegree>,
     // - divide by number of rows (since we're doing an inverse DFT)
     // - multiply by powers of the coset shift (see default coset LDE impl for
     // an explanation)
-    std::vector<F> weights =
-        F::GetSuccessivePowers(this->size_, shift, this->size_inv_);
-    OMP_PARALLEL_FOR(size_t row = 0; row < weights.size(); ++row) {
+    base::Parallelize(this->size_, [this, &mat, &shift](size_t len,
+                                                        size_t chunk_offset,
+                                                        size_t chunk_size) {
       // Reverse bits because |mat| is encoded in bit-reversed order
-      mat.row(base::bits::ReverseBitsLen(row, this->log_size_of_group_)) *=
-          weights[row];
-    }
+      size_t start = chunk_offset * chunk_size;
+      F weight = this->size_inv_ * shift.Pow(start);
```

Comment:
Since we do more multiplication, don't we lose performance by doing like this?

Reply:
Technically, it multiplies `thread_size` - 1 times more due. But I believe it doesn't lose performance, rather, it increase by avoid heap allocation and reducing the number of thread join.

Reply:
I believe multiplication is more than that. We do more `thread_size` - 1 times more Pow.

Reply:
Oh no, actually the number of the multiplications are same, if you see `GetSuccessivePowers()` in `tachyon/math/base/semigroups.h`, the way to compute powers are same!